webapp/src/components/emptyCenterPanel.tsx

```diff
@@ -1,34 +1,170 @@
+import BoardIcon from '../widgets/icons/board'
+import octoClient from '../octoClient'
+
+import {addBoardTemplateClicked, addBoardClicked} from './sidebar/sidebarAddBoardMenu'
+import {addBoardFromTemplate, BoardTemplateMenuItemOptionsButton} from './sidebar/boardTemplateMenuItem'
+
 import './emptyCenterPanel.scss'
 
-const EmptyCenterPanel = React.memo(() => {
-    const workspace = useAppSelector(getCurrentWorkspace)
+type ButtonProps = {
+    buttonIcon: string | React.ReactNode,
+    title: string,
+    readonly: boolean,
+    onClick: () => void,
+    showBoard?: (boardId: string) => void
+    boardTemplate?: Board
+}
+
+const EmptyCenterPanelButton = React.memo((props: ButtonProps) => {
+    const {onClick, buttonIcon, title, readonly, showBoard, boardTemplate} = props
 
     return (
-        <div className='EmptyCenterPanel'>
-            {workspace && workspace.id !== '0' &&
-            <div className='WorkspaceInfo'>
-                <FormattedMessage
-                    id='EmptyCenterPanel.workspace'
-                    defaultMessage='This is the workspace for:'
+        <div
+            onClick={onClick}
+            className='button'
+        >
+            <i>{buttonIcon}</i>
+            <span className='button-title'>{title}</span>
+            {!readonly &&
+                <BoardTemplateMenuItemOptionsButton
+                    showBoard={showBoard!}
+                    boardTemplate={boardTemplate!}
                 />
-                <b>
-                    {workspace.title}
-                </b>
-            </div>
             }
-            <div className='Hint'>
+        </div>
+    )
+})
+
+const EmptyCenterPanel = React.memo(() => {
+    const workspace = useAppSelector(getCurrentWorkspace)
+    const templates = useAppSelector(getSortedTemplates)
+    const globalTemplates = useAppSelector<Board[]>(getGlobalTemplates)
+    const history = useHistory()
+    const dispatch = useAppDispatch()
+    const intl = useIntl()
+    const match = useRouteMatch<{boardId: string, viewId?: string}>()
+
+    useEffect(() => {
+        if (octoClient.workspaceId !== '0' && globalTemplates.length === 0) {
+            dispatch(fetchGlobalTemplates())
+        }
+    }, [octoClient.workspaceId])
+
+    const showBoard = useCallback((boardId) => {
+        const params = {...match.params, boardId: boardId || ''}
+        delete params.viewId
+        const newPath = generatePath(match.path, params)
+        history.push(newPath)
+    }, [match, history])
+
+    const newTemplateClicked = () => addBoardTemplateClicked(showBoard, intl)
+    const emptyBoardClicked = () => addBoardClicked(showBoard, intl)
+
+    let contentDisplay = (
+        <div className='Hint'>
+            <FormattedMessage
+                id='EmptyCenterPanel.no-content'
+                defaultMessage='Add or select a board from the sidebar to get started.'
+            />
+        </div>
+    )
+
+    if (Utils.isFocalboardPlugin()) {
```

Comment:
It's not a nice practice to have the majority of the logic in conditional blocks.

This can be simplified as follows-

```javascript
    if (!Utils.isFocalboardPlugin()) {
        return (
            <div className='EmptyCenterPanel'>
                <div className='Hint'>
                    <FormattedMessage
                        id='EmptyCenterPanel.no-content'
                        defaultMessage='Add or select a board from the sidebar to get started.'
                    />
                </div>
            </div>
        )
    }

    return (...contents of this if block)
```

Reply:
Smart 💯 Thanks. Done